### ISSUE TITLE
Reward Models: call for contributions

### ISSUE THREAD

**Issue by s1lent4gnt** [huggingface/lerobot#3143]
# Reward Models Refactoring: Call for Contributions

## Context

LeRobot now has a dedicated reward models foundation (`src/lerobot/rewards/`) introduced in PR #3142. The goal is to make **adding new reward models** as easy as adding a new policy — one model, one config file — and to support reward signals across IL (sample weighting), RL (online reward shaping), and evaluation.

Two reward models already work through this foundation:
- **Reward Classifier**
- **SARM**

Now we want the community's help to **expand the reward model ecosystem** and **tighten the integration** with the rest of LeRobot.

---

## Current Architecture

This architecture is introduced in the https://github.com/huggingface/lerobot/pull/3142. The Reward models lives in src/lerobot/rewards/ and follows LeRobot's existing patterns:
```
src/lerobot/
├── configs/rewards.py           # RewardModelConfig (draccus.ChoiceRegistry + HubMixin)
├── rewards/
│   ├── __init__.py              # Registry population (eager config imports)
│   ├── pretrained.py            # PreTrainedRewardModel ABC (nn.Module + HubMixin)
│   ├── factory.py               # get_reward_model_class, make_reward_model, make_reward_pre_post_processors
│   ├── classifier/              # Reward Classifier (trainable)
│   └── sarm/                    # SARM (trainable)
```

**Key design decisions already in place:**

- `RewardModelConfig` uses `draccus.ChoiceRegistry` — same registry pattern as policies
- `PreTrainedRewardModel` has two APIs: `compute_reward()` (inference) and `forward()` (training, optional)
- Hub integration via `HubMixin` — save/load/push to Hugging Face Hub

---

### Phase 1: Improve existing infrastructure

- Better error messages and validation in `PreTrainedRewardModel`
- Processor pipelines for reward models
- Documentation and tutorials
- Benchmarks comparing reward models on standard tasks
- Add `reward_model: RewardModelConfig | None` field to `TrainPipelineConfig`
- Separate policy training and reward model training paths at the config level
- CLI: `lerobot-train --reward_model.type=sarm --dataset.repo_id=...`

### Phase 2: New Reward Models

- **ReWiND** — reward from video prediction

Adding a new reward model requires:
1. `configuration_*.py` — config dataclass inheriting `RewardModelConfig`
2. `modeling_*.py` — model class inheriting `PreTrainedRewardModel`
3. Register with `@RewardModelConfig.register_subclass("your_model_name")`

---

### Share ideas

- What reward model approaches have worked best for your tasks?
- What's missing from the current API?
- What would make reward model integration easier in your workflow?

---

## Open for Ideas

Please **comment below** or **open a PR**. Contributions of any size are welcome — a comment, a bug report, or a full implementation.

**Comment by 0xPraedico** [comment]
Hi @s1lent4gnt 
I just opened a small Phase-1 contribution here: #3192.

It focuses on DevEx in the reward factory (clearer unknown-type errors + tests).  
I’d be happy to contribute more on this issue.

If there is already a plan (or if we should define one), I’d be glad to discuss it on Discord: nord5530  
or via Twitter DM: praedico

**Comment by 0xPraedico** [comment]
I just opened a PR for VITA integration in the new reward-model foundation: #3269  Paper: [arXiv:2506.10085](https://arxiv.org/pdf/2506.10085)
It includes the core VITA pipeline (`z_t`, `P_K/P_V/P_Q`, fast adaptation, sequential updates, reward head), + tests and docs.
Happy to iterate on scope for follow-ups (native VLM backbone / paper sampling strategy)

**Comment by int-smart** [comment]
Hi @s1lent4gnt 
Added the RoboReward model. Please review and let me know if changes are required.

**Comment by philipmit** [comment]
I added the Robometer model! Let me know if any changes are needed.

**Comment by JeremyHX** [comment]
We've contributed the IL (sample-weighting) axis of this call, in two small PRs:

1. **`ACTPolicy.forward(reduction="none")`** — #4221 — today `update_policy` calls `policy.forward(batch, reduction="none")` whenever `sample_weighting` is configured, and `_make_rabc_weighter`'s error message names ACT as an intended target — but ACT doesn't accept the argument, so RA-BC + ACT crashes with a TypeError before the first step (repro in the PR). The fix mirrors the SmolVLA implementation (#2639): per-sample masked mean with the same normalization as the scalar path; default path unchanged.
2. **A `type: "static"` sample weighter** — #4222 — precomputed per-frame weights from a parquet (`index` = global frame index, `weight` ≥ 0), keyed on the batch's `index` exactly like `RABCWeights`, batch-normalized to sum to batch size. This is the generic half of the abstraction's stated motivation (importance sampling / curriculum / quality filtering, #2781): any offline scorer — data-quality classifiers, DAgger human-vs-policy provenance, curriculum schedules — writes one file and trains with it, no new weighter class needed. Zero training-loop changes.

Our use case: HIL-DAgger corrections on an SO-101 cucumber-prep rig — we stamp per-frame human/policy provenance at recording time and want corrections weighted above policy self-play without forking the trainer. Happy to rename "static" (e.g. "precomputed") if preferred — feedback welcome on either PR.
